lnpbp-0002.md

```diff
@@ -112,29 +112,27 @@ from a given Bitcoin script, based on Miniscript [16].
 ### Commitment procedure
 
 The **committing party**, having a message `msg`, must:
-1. Collect all unique public key instances (named **original public key set** 
-   `P`) related to a given transaction output, defined as:
-   - a single public key for P2PK, P2PKH, P2WPK, P2WPK-in-P2SH type of outputs;
-   - an arbitrary public key (even with an unknown corresponding private key),
-     if `OP_RETURN` type of `scriptPubkey` is used;
-   - an intermediate public key for the Taproot-based (SegWit v1) output;
-   - all public keys extracted using 
-     [algorithm](#deterministic-public-key-extraction-from-bitcoin-script) from 
-     either:
-     * `witnessScript` contained in the `witness` stack for native v0 P2WSH
-       and legacy P2WSH-in-P2SH SegWit outputs;
-     * `redeemScript` contained in top  `scriptSig` for non-SegWit P2SH
-     * or bare (custom) script-based `scriptPubkey`
-2. Select a single public key `Po` from the set `P` of the original public keys 
-   for the tweaking procedure. It is advised that the corresponding private key
-   being controlled by the committing party, which will simplify future spending
-   of the output.
-3. Run [LNPBP-1 commitment procedure](https://github.com/LNP-BP/LNPBPs/blob/master/lnpbp-0001.md#commitment-procedure)
-   for the message `msg`, selected public key `Po`, public key set `P` and a
-   protocol-specific `tag`, provided by the upstream protocol using this 
-   standard.
+1. Collect all public keys instances (named **original public key set**) 
+   related to the given transaction output, defined as:
+   - a single public key for P2PK, P2PKH, P2WPK, P2WPK/WSH-in-P2SH type of outputs;
+   - an arbitrary public key (even with an unknown corresponding private key), 
+     if `OP_RETURN` `scriptPubkey` is used;
+   - an internal public key for a Taproot output (P2TR);
+   - all public keys from a `redeemScript` (for P2(W)SH and P2WSH-in-P2SH, 
```

Comment:
Since a public key may appear in a script multiple times I think its important to state that the set must contain a single copy of each key only, i.e. the keys should be unique. It was stated before in the pt.1, but I think this can be moved to this subpoint, which is the only variant where multiple public keys appear

Reply:
I don't know, after reading this part and the `Deterministic public key extraction from Bitcoin Script` it refers to, I feel that it's already pretty clear in the 3rd step of public key extraction: 
```
Select unique public keys (i.e. if some public key is repeated in different parts of the script/in different script branches, pick a single instance of it).
```
If we want to mention that here too, I guess that adding `unique` to l.121 would be enough, making it `all unique public key`.

Reply:
Ok. Overall I think more clarification is better, but I think we do not need to spend more time on this. Merging now